### PR TITLE
Remove Parameters from interface with oops

### PR DESCRIPTION
**Pull request by ytremolet**
## Description

See jcsda-internal/oops#2507

## Dependencies

build-group=https://github.com/JCSDA-internal/oops#2507
build-group=https://github.com/JCSDA-internal/ioda#1186
build-group=https://github.com/JCSDA-internal/ufo#3162
build-group=https://github.com/JCSDA-internal/fv3-jedi#1116

### CODE REVIEW

File: src/soca/AnalyticInit/AnalyticInit.h [hunk 1]
```diff
@@ -8,13 +8,19 @@
...
 namespace soca {
 
-  class AnalyticInitParameters : public oops::AnalyticInitParametersBase {
-    OOPS_CONCRETE_PARAMETERS(AnalyticInitParameters, AnalyticInitParametersBase)
+  class AnalyticInitParameters : public oops::Parameters {
+    OOPS_CONCRETE_PARAMETERS(AnalyticInitParameters, Parameters)
+    oops::RequiredParameter<std::string> method{"method", this};
    public:
     // No parameters, for now
   };
```
Comment (fmahebert) [inline src/soca/AnalyticInit/AnalyticInit.h]: Can this class `AnalyticInitParameters` be deleted (and the parameters-related includes too) now that you're passing a config only to the constructor below?

**Review by fmahebert — COMMENTED** [review]
I wonder if the CI is failing because build-group links are malformed (try clicking on one link — they're not the actual URL to the pull request, the link should be like `repo/pull/number` not `repo#number`)

One question:

File: src/soca/AnalyticInit/AnalyticInit.h (hunk 1 above)
Comment (ytremolet) [reply src/soca/AnalyticInit/AnalyticInit.h]: yes

Comment (travissluka) [reply src/soca/AnalyticInit/AnalyticInit.h]: I think that whole AnalyticInit can go away, I don't think it's used anywhere anymore.... deal with that another day.

**Review by travissluka — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Review by srherbener — APPROVED** [review]

**Pull request merged**